discovery_engine_flutter/example/lib/main.dart

```diff
@@ -114,17 +115,21 @@ class _MyAppState extends State<MyApp> {
             mainAxisSize: MainAxisSize.min,
             children: [
               if (engineState == EngineState.initializing)
-                Text('Fetching assets: ${progress.toStringAsPrecision(2)}%'),
-              if (engineState == EngineState.ready)
+                Text('Fetching assets: ${progress.toStringAsFixed(0)}%'),
+              if (engineState == EngineState.ready) ...[
                 const Text('Engine initialized'),
-              if (engineState != EngineState.ready)
                 ElevatedButton(
-                  onPressed:
-                      engineState == EngineState.notReady ? initEngine : null,
-                  child: engineState == EngineState.notReady
-                      ? const Text('Init engine')
-                      : const Text('Initializing...'),
+                  onPressed: requestNews,
+                  child: const Text('Request News'),
+                ),
+              ],
+              if (engineState == EngineState.notReady)
+                ElevatedButton(
+                  onPressed: initEngine,
```

Comment:
Can we automatically call `initEngine` when build is called. Also fetch one batch automatically.

It will fast exit if it was called before.

And there is no reason to wait for a user interaction.

I did that change ad-hoc on all branches where I tested things regarding the compiler bug.

Reply:
We also probably still want to call `await engine.requestFeed();` in `initEngine`.

As a side note, we probably call it `restoreFeed`. I found it being named `requestFeed` prone to confusing with `requestNextFeedBatch`.

Reply:
I've dropped the need for user interaction, added `requestFeed`.

I agree we should rename it to `restoreFeed`, but maybe let's do it after the beta release, cause it's a breaking change, wdyt?

Reply:
We also talked with Frank about this renaming, we can do it after the release